### PR TITLE
Use Tzdata for timezones

### LINKED ISSUE #904: Timezone list doesn't follow the time change
"The UK is currently on BST" which is GMT+1. CET is now GMT+2 etc while our static timezone list still has the previous time. Like I mentioned on Twitter, this doesn't affect the data/charts but is more cosmetic that should be fixed.

ExwiM9qWQAEbfAL

([feedback via twitter](https://twitter.com/ChildishGiant/status/1377013495909203968))

### LINKED ISSUE #1340: Timezone list breaks with certain IANA valid timezones
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Describe the bug

1. When creating a site through the API and setting the timezone to `Europe/Berlin`
2. The dashboard works fine and reporting is done in the correct timezone
3. But the timezone dropdown in site settings shows GMT-12 Internation Date Line West which is clearly wrong

### Expected behavior

Shows the timezone as `Europe/Berlin`

### Screenshots

_No response_

### Environment

_No response_

### PR DESCRIPTION
**Pull request by vinibrsl**
### Changes

This pull request fixes the following issues with timezones:
- #904 
- #1340 
- https://github.com/plausible/analytics/issues/904#issuecomment-811093097

I used [Tzdata](https://github.com/lau/tzdata) to generate the timezone list. The library was extracted from the built-in Elixir.Calendar module. This includes missing timezones as Europe/Berlin and follows time updates accordingly.

Additionally, I improved the timezone recommendation when creating a new site. The current solution uses the TZ offset to match a timezone. This solution works for some cases, but different TZ can have the same offset. This PR uses the JavaScript `Intl.DateTimeFormat` API when available. This API returns the timezone name, instead of the offset, which is now a fallback.

### Tests
- [X] Automated tests have been added

### Changelog
- [X] Entry has been added to changelog

### Documentation
- [X] This change does not need a documentation update

### Dark mode
- [X] This PR does not change the UI

Closes #904, closes #1340

### CODE REVIEW

**Comment by metmarkosaric** [comment]
oh wow this will be a great improvement! thanks @vinibrsl!

**Comment by ukutaht** [comment]
Looks good @vinibrsl ! There's a formatting error but otherwise ready to merge

**Comment by vinibrsl** [comment]
> Nice!
> 
> > Closes #904, #1340
> 
> Seems like #1340 isn't linked to this PR properly, I think you have to write it separately like this:
> 
> Closes #904 Closes #1340

You're right! Thanks for pointing that out. I updated the PR description.

File: lib/plausible_web/templates/site/new.html.eex [hunk 1]
```diff
@@ -63,8 +63,18 @@
...
-      var offset = (new Date()).getTimezoneOffset()
-      var option = document.querySelector('#tz-select option[offset="' + offset + '"]')
+      var option;
+
+      if (typeof Intl !== "undefined") {
+        var timezoneName = Intl.DateTimeFormat().resolvedOptions().timeZone
+        option = document.querySelector('#tz-select option[value="' + timezoneName + '"]')
+      }
+
+      if (!option) {
```
Comment (ukutaht) [inline lib/plausible_web/templates/site/new.html.eex:73]: Why would this be the case? Missing `Intl` on old browser versions?

**Review by ukutaht — APPROVED** [review]
Looks great! Just one question

File: lib/plausible_web/templates/site/new.html.eex (hunk 1 above)
Comment (vinibrsl) [reply lib/plausible_web/templates/site/new.html.eex:73]: Yes. In cases where Intl is not available or it couldn't fetch the user TZ for some reason.

**Comment by vinibrsl** [comment]
Are we good to merge this @ukutaht?

**Comment by vinibrsl** [comment]
Closing in favor of #2222

**Pull request closed without merging**